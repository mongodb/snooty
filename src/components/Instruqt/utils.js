export const getLabURL = (embedValue) => {
  return `https://play.instruqt.com/embed${embedValue}`;
};
