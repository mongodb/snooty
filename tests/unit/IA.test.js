import React from 'react';
import { render } from '@testing-library/react';
// Keep mockLocation on top to ensure mock is applied
import { mockLocation } from '../utils/mock-location';
import IA from '../../src/components/Sidenav/IA';
import useSnootyMetadata from '../../src/utils/use-snooty-metadata';
import sampleData from './data/IA.test.json';

jest.mock(`../../src/utils/use-snooty-metadata`, () => jest.fn());

beforeAll(() => {
  mockLocation(null, `/`);
});

it('renders a simple page IA correctly', () => {
  useSnootyMetadata.mockImplementation(() => ({}));
  const tree = render(<IA ia={sampleData.pageIA} />);
  expect(tree.asFragment()).toMatchSnapshot();
});

it('renders a page IA with IA linked data', () => {
  useSnootyMetadata.mockImplementation(() => sampleData.iaTreeMetadata);
  const mockIAData = [...sampleData.pageIA];
  // Programmatically add an ID for linked data, in case sample data order changes.
  for (const mockData of mockIAData) {
    if (mockData.title[0].value === 'Client Libraries') {
      mockData.id = 'client-libraries';
      break;
    }
  }
  const tree = render(<IA ia={mockIAData} />);
  expect(tree.asFragment()).toMatchSnapshot();
  // Explicitly check for a piece of linked data. The rest should be visible in
  // the snapshot.
  expect(tree.getByText('C++')).toBeTruthy();
});
