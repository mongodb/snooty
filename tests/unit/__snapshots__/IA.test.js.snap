// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a page IA with IA linked data 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
}

.emotion-0~.emotion-0>.lg-ui-side-nav-group-button-0000 {
  padding: 16px 16px 8px 16px;
}

.emotion-0>div {
  padding: 0!important;
}

.emotion-1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  margin-top: 0;
  margin-bottom: 0;
  padding: 16px 16px 8px;
  color: #00684A;
}

.emotion-2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 8px;
}

.emotion-3 {
  margin: unset;
  font-family: 'Euclid Circular A','Helvetica Neue',Helvetica,Arial,sans-serif;
  color: #001E2B;
  font-size: 12px;
  font-weight: 700;
  text-transform: uppercase;
  line-height: 20px;
  letter-spacing: 0.4px;
  color: #001E2B;
  color: inherit;
}

.emotion-4 {
  margin-block-start: 0px;
  margin-block-end: 0px;
  -webkit-padding-start: 0px;
  padding-inline-start: 0px;
  padding: 0;
  list-style-type: none;
}

.emotion-5 {
  width: 100%;
  list-style: none;
}

.emotion-6 {
  font-family: 'Euclid Circular A','Helvetica Neue',Helvetica,Arial,sans-serif;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-decoration-color: transparent;
  cursor: pointer;
  font-size: inherit;
  line-height: inherit;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background: none;
  border: none;
  padding: 0;
  font-size: 13px;
  line-height: 20px;
  color: #016BF8;
  font-weight: 400;
  display: inline;
  margin: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background: none;
  border: none;
  cursor: pointer;
  position: relative;
  width: 100%;
  min-height: 32px;
  padding: 6px 16px;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: 'Euclid Circular A','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: 400;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: capitalize;
  -webkit-transition: background-color 100ms ease-in-out;
  transition: background-color 100ms ease-in-out;
  color: #001E2B;
  background-color: rgba(249,251,250,0);
  font-size: 13px;
  line-height: 20px;
  position: relative;
  padding-left: 24px;
  padding-right: 24px;
  padding-top: 8px;
  padding-bottom: 8px;
  font-size: 13px;
  line-height: 20px;
}

.emotion-6:hover,
.emotion-6:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-transition: text-decoration 150ms ease-in-out;
  transition: text-decoration 150ms ease-in-out;
  text-underline-offset: 4px;
  text-decoration-thickness: 2px;
}

.emotion-6:focus {
  outline: none;
}

.emotion-6:hover {
  text-decoration-color: #E8EDEB;
}

.emotion-6:focus {
  text-decoration-color: #016BF8;
}

.emotion-6:hover {
  background-color: #E8EDEB;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-6:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: none;
}

.emotion-6::-moz-focus-inner {
  border: 0;
}

.emotion-6:before {
  content: '';
  position: absolute;
  background-color: transparent;
  left: 0;
  top: 6px;
  bottom: 6px;
  width: 4px;
  border-radius: 0 6px 6px 0;
  -webkit-transition: -webkit-transform 150ms ease-in-out;
  transition: transform 150ms ease-in-out;
  -webkit-transform: scaleY(0.3);
  -moz-transform: scaleY(0.3);
  -ms-transform: scaleY(0.3);
  transform: scaleY(0.3);
}

.emotion-6:hover {
  background-color: #E8EDEB;
}

.emotion-6:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-6:focus:before {
  -webkit-transform: scaleY(1);
  -moz-transform: scaleY(1);
  -ms-transform: scaleY(1);
  transform: scaleY(1);
}

.emotion-6:focus {
  color: #083C90;
  background-color: #E1F7FF;
}

.emotion-6:focus:before {
  background-color: #016BF8;
}

.emotion-10 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-decoration-color: transparent;
  line-height: 13px;
  color: var(--color);
  font-weight: var(--font-weight);
}

.emotion-10>code {
  color: var(--color);
}

.emotion-10:focus,
.emotion-10:hover {
  text-decoration-line: underline;
  -webkit-transition: text-decoration 150ms ease-in-out;
  transition: text-decoration 150ms ease-in-out;
  text-underline-offset: 4px;
  text-decoration-thickness: 2px;
}

.emotion-10:focus {
  text-decoration-color: var(--focus-text-decoration-color);
  outline: none;
}

.emotion-10:hover {
  text-decoration-color: var(--hover-text-decoration-color);
}

.emotion-11 {
  margin: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background: none;
  border: none;
  cursor: pointer;
  position: relative;
  width: 100%;
  min-height: 32px;
  padding: 6px 16px;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: 'Euclid Circular A','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: 400;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: capitalize;
  -webkit-transition: background-color 100ms ease-in-out;
  transition: background-color 100ms ease-in-out;
  color: #001E2B;
  background-color: rgba(249,251,250,0);
  font-size: 13px;
  line-height: 20px;
  position: relative;
  padding-left: 24px;
  padding-right: 24px;
  padding-top: 8px;
  padding-bottom: 8px;
  font-size: 13px;
  line-height: 20px;
}

.emotion-11:hover {
  background-color: #E8EDEB;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-11:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: none;
}

.emotion-11::-moz-focus-inner {
  border: 0;
}

.emotion-11:before {
  content: '';
  position: absolute;
  background-color: transparent;
  left: 0;
  top: 6px;
  bottom: 6px;
  width: 4px;
  border-radius: 0 6px 6px 0;
  -webkit-transition: -webkit-transform 150ms ease-in-out;
  transition: transform 150ms ease-in-out;
  -webkit-transform: scaleY(0.3);
  -moz-transform: scaleY(0.3);
  -ms-transform: scaleY(0.3);
  transform: scaleY(0.3);
}

.emotion-11:hover {
  background-color: #E8EDEB;
}

.emotion-11:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-11:focus:before {
  -webkit-transform: scaleY(1);
  -moz-transform: scaleY(1);
  -ms-transform: scaleY(1);
  transform: scaleY(1);
}

.emotion-11:focus {
  color: #083C90;
  background-color: #E1F7FF;
}

.emotion-11:focus:before {
  background-color: #016BF8;
}

.emotion-14 {
  display: grid;
  -webkit-column-gap: 8px;
  column-gap: 8px;
  row-gap: 4px;
  grid-template-columns: 1fr 1fr;
  list-style-type: none;
  -webkit-padding-start: 0px;
  padding-inline-start: 0px;
  padding: 8px;
}

.emotion-16 {
  font-family: 'Euclid Circular A','Helvetica Neue',Helvetica,Arial,sans-serif;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-decoration-color: transparent;
  cursor: pointer;
  font-size: inherit;
  line-height: inherit;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background: none;
  border: none;
  padding: 0;
  font-size: 13px;
  line-height: 20px;
  color: #016BF8;
  font-weight: 400;
  display: inline;
  margin: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background: none;
  border: none;
  cursor: pointer;
  position: relative;
  width: 100%;
  min-height: 32px;
  padding: 6px 16px;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: 'Euclid Circular A','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: 400;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: capitalize;
  -webkit-transition: background-color 100ms ease-in-out;
  transition: background-color 100ms ease-in-out;
  color: #001E2B;
  background-color: rgba(249,251,250,0);
  font-size: 13px;
  line-height: 20px;
  position: relative;
  border-radius: 20px;
  font-size: 13px;
  max-width: 120px;
  height: 32px;
}

.emotion-16:hover,
.emotion-16:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-transition: text-decoration 150ms ease-in-out;
  transition: text-decoration 150ms ease-in-out;
  text-underline-offset: 4px;
  text-decoration-thickness: 2px;
}

.emotion-16:focus {
  outline: none;
}

.emotion-16:hover {
  text-decoration-color: #E8EDEB;
}

.emotion-16:focus {
  text-decoration-color: #016BF8;
}

.emotion-16:hover {
  background-color: #E8EDEB;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-16:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: none;
}

.emotion-16::-moz-focus-inner {
  border: 0;
}

.emotion-16:before {
  content: '';
  position: absolute;
  background-color: transparent;
  left: 0;
  top: 6px;
  bottom: 6px;
  width: 4px;
  border-radius: 0 6px 6px 0;
  -webkit-transition: -webkit-transform 150ms ease-in-out;
  transition: transform 150ms ease-in-out;
  -webkit-transform: scaleY(0.3);
  -moz-transform: scaleY(0.3);
  -ms-transform: scaleY(0.3);
  transform: scaleY(0.3);
}

.emotion-16:hover {
  background-color: #E8EDEB;
}

.emotion-16:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-16:focus:before {
  -webkit-transform: scaleY(1);
  -moz-transform: scaleY(1);
  -ms-transform: scaleY(1);
  transform: scaleY(1);
}

.emotion-16:focus {
  color: #083C90;
  background-color: #E1F7FF;
}

.emotion-16:focus:before {
  background-color: #016BF8;
}

.emotion-16:focus {
  border: 1px solid #0498EC;
}

.emotion-16:focus:focus:before {
  display: none;
}

.emotion-16:focus>span>span {
  position: relative;
}

.emotion-16:focus>span>span::after {
  content: '';
  position: absolute;
  width: 100%;
  height: 1px;
  bottom: 4px;
  left: 0;
  border-radius: 2px;
  background-color: #0498EC;
}

.emotion-16>span {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 8px;
}

.emotion-16>span:after {
  display: none;
}

<li
    class="emotion-0"
  >
    <div
      class="emotion-1"
      data-testid="side-nav-group-header-label"
      id="menu-group-label-id-2"
    >
      <div
        class="emotion-2"
      >
        <div
          class="emotion-3"
        />
      </div>
    </div>
    <ul
      aria-labelledby="menu-group-label-id-2"
      class="emotion-4"
    >
      <li
        class="emotion-5"
      >
        <a
          aria-current="false"
          aria-disabled="false"
          class="lg-ui-0001 lg-ui-side-nav-item-0000 emotion-6"
          href="https://www.mongodb.com/docs/atlas/getting-started/"
          target="_self"
        >
          <span>
            Atlas
          </span>
        </a>
      </li>
      <li
        class="emotion-5"
      >
        <a
          aria-current="false"
          aria-disabled="false"
          class="lg-ui-0001 lg-ui-side-nav-item-0000 emotion-6"
          href="https://www.mongodb.com/docs/manual/"
          target="_self"
        >
          <span>
            MongoDB Database Manual
          </span>
        </a>
      </li>
      <li
        class="emotion-5"
      >
        <a
          aria-current="false"
          aria-disabled="false"
          class="emotion-10 lg-ui-side-nav-item-0000 emotion-11"
          href="/tools-and-connectors/"
          style="--color: #016BF8; --focus-text-decoration-color: #016BF8; --hover-text-decoration-color: #E8EDEB; --font-weight: inherit;"
        >
          Tools and Connectors
        </a>
      </li>
      <li
        class="emotion-5"
      >
        <a
          aria-current="false"
          aria-disabled="false"
          class="lg-ui-0001 lg-ui-side-nav-item-0000 emotion-6"
          href="https://www.mongodb.com/docs/drivers/"
          target="_self"
        >
          <span>
            Client Libraries
          </span>
        </a>
      </li>
      <ul
        class="emotion-14"
      >
        <li
          class="emotion-5"
        >
          <a
            aria-current="false"
            aria-disabled="false"
            class="lg-ui-0001 lg-ui-side-nav-item-0000 emotion-16"
            href="https://www.mongodb.com/docs/drivers/c/"
            target="_self"
          >
            <span>
              <img
                alt="C Driver icon"
                height="16"
                src="/icons/c.svg"
                width="16"
              />
              <span>
                C
              </span>
            </span>
          </a>
        </li>
        <li
          class="emotion-5"
        >
          <a
            aria-current="false"
            aria-disabled="false"
            class="lg-ui-0001 lg-ui-side-nav-item-0000 emotion-16"
            href="https://www.mongodb.com/docs/drivers/cxx/"
            target="_self"
          >
            <span>
              <img
                alt="C plus plus Driver icon"
                height="16"
                src="/icons/c++.svg"
                width="16"
              />
              <span>
                C++
              </span>
            </span>
          </a>
        </li>
        <li
          class="emotion-5"
        >
          <a
            aria-current="false"
            aria-disabled="false"
            class="lg-ui-0001 lg-ui-side-nav-item-0000 emotion-16"
            href="https://www.mongodb.com/docs/drivers/csharp/current/"
            target="_self"
          >
            <span>
              <img
                alt="C sharp Driver icon"
                height="16"
                src="/icons/csharp.svg"
                width="16"
              />
              <span>
                C#
              </span>
            </span>
          </a>
        </li>
        <li
          class="emotion-5"
        >
          <a
            aria-current="false"
            aria-disabled="false"
            class="lg-ui-0001 lg-ui-side-nav-item-0000 emotion-16"
            href="https://www.mongodb.com/docs/drivers/go/current/"
            target="_self"
          >
            <span>
              <img
                alt="Go Driver icon"
                height="16"
                src="/icons/go.svg"
                width="16"
              />
              <span>
                Go
              </span>
            </span>
          </a>
        </li>
        <li
          class="emotion-5"
        >
          <a
            aria-current="false"
            aria-disabled="false"
            class="lg-ui-0001 lg-ui-side-nav-item-0000 emotion-16"
            href="https://www.mongodb.com/docs/drivers/java-drivers/"
            target="_self"
          >
            <span>
              <img
                alt="Java Driver icon"
                height="16"
                src="/icons/java.svg"
                width="16"
              />
              <span>
                Java
              </span>
            </span>
          </a>
        </li>
        <li
          class="emotion-5"
        >
          <a
            aria-current="false"
            aria-disabled="false"
            class="lg-ui-0001 lg-ui-side-nav-item-0000 emotion-16"
            href="https://www.mongodb.com/docs/drivers/kotlin-drivers/"
            target="_self"
          >
            <span>
              <img
                alt="Kotlin Driver icon"
                height="16"
                src="/icons/kotlin.svg"
                width="16"
              />
              <span>
                Kotlin
              </span>
            </span>
          </a>
        </li>
        <li
          class="emotion-5"
        >
          <a
            aria-current="false"
            aria-disabled="false"
            class="lg-ui-0001 lg-ui-side-nav-item-0000 emotion-16"
            href="https://www.mongodb.com/docs/drivers/node/current/"
            target="_self"
          >
            <span>
              <img
                alt="Node Driver icon"
                height="16"
                src="/icons/node.svg"
                width="16"
              />
              <span>
                Node.js
              </span>
            </span>
          </a>
        </li>
        <li
          class="emotion-5"
        >
          <a
            aria-current="false"
            aria-disabled="false"
            class="lg-ui-0001 lg-ui-side-nav-item-0000 emotion-16"
            href="https://www.mongodb.com/docs/drivers/php/"
            target="_self"
          >
            <span>
              <img
                alt="PHP Driver icon"
                height="16"
                src="/icons/php.svg"
                width="16"
              />
              <span>
                PHP
              </span>
            </span>
          </a>
        </li>
        <li
          class="emotion-5"
        >
          <a
            aria-current="false"
            aria-disabled="false"
            class="lg-ui-0001 lg-ui-side-nav-item-0000 emotion-16"
            href="https://www.mongodb.com/docs/drivers/python/"
            target="_self"
          >
            <span>
              <img
                alt="Python Driver icon"
                height="16"
                src="/icons/python.svg"
                width="16"
              />
              <span>
                Python
              </span>
            </span>
          </a>
        </li>
        <li
          class="emotion-5"
        >
          <a
            aria-current="false"
            aria-disabled="false"
            class="lg-ui-0001 lg-ui-side-nav-item-0000 emotion-16"
            href="https://www.mongodb.com/docs/drivers/ruby/"
            target="_self"
          >
            <span>
              <img
                alt="Ruby Driver icon"
                height="16"
                src="/icons/ruby.svg"
                width="16"
              />
              <span>
                Ruby
              </span>
            </span>
          </a>
        </li>
        <li
          class="emotion-5"
        >
          <a
            aria-current="false"
            aria-disabled="false"
            class="lg-ui-0001 lg-ui-side-nav-item-0000 emotion-16"
            href="https://www.mongodb.com/docs/drivers/rust/"
            target="_self"
          >
            <span>
              <img
                alt="Rust Driver icon"
                height="16"
                src="/icons/rust.svg"
                width="16"
              />
              <span>
                Rust
              </span>
            </span>
          </a>
        </li>
        <li
          class="emotion-5"
        >
          <a
            aria-current="false"
            aria-disabled="false"
            class="lg-ui-0001 lg-ui-side-nav-item-0000 emotion-16"
            href="https://www.mongodb.com/docs/drivers/scala/"
            target="_self"
          >
            <span>
              <img
                alt="Scala Driver icon"
                height="16"
                src="/icons/scala.svg"
                width="16"
              />
              <span>
                Scala
              </span>
            </span>
          </a>
        </li>
        <li
          class="emotion-5"
        >
          <a
            aria-current="false"
            aria-disabled="false"
            class="lg-ui-0001 lg-ui-side-nav-item-0000 emotion-16"
            href="https://www.mongodb.com/docs/drivers/swift/"
            target="_self"
          >
            <span>
              <img
                alt="Swift Driver icon"
                height="16"
                src="/icons/swift.svg"
                width="16"
              />
              <span>
                Swift
              </span>
            </span>
          </a>
        </li>
        <li
          class="emotion-5"
        >
          <a
            aria-current="false"
            aria-disabled="false"
            class="lg-ui-0001 lg-ui-side-nav-item-0000 emotion-16"
            href="https://www.mongodb.com/docs/drivers/typescript/"
            target="_self"
          >
            <span>
              <img
                alt="TypeScript Driver icon"
                height="16"
                src="/icons/ts.svg"
                width="16"
              />
              <span>
                TypeScript
              </span>
            </span>
          </a>
        </li>
      </ul>
    </ul>
  </li>
</DocumentFragment>
`;

exports[`renders a simple page IA correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
}

.emotion-0~.emotion-0>.lg-ui-side-nav-group-button-0000 {
  padding: 16px 16px 8px 16px;
}

.emotion-0>div {
  padding: 0!important;
}

.emotion-1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  margin-top: 0;
  margin-bottom: 0;
  padding: 16px 16px 8px;
  color: #00684A;
}

.emotion-2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 8px;
}

.emotion-3 {
  margin: unset;
  font-family: 'Euclid Circular A','Helvetica Neue',Helvetica,Arial,sans-serif;
  color: #001E2B;
  font-size: 12px;
  font-weight: 700;
  text-transform: uppercase;
  line-height: 20px;
  letter-spacing: 0.4px;
  color: #001E2B;
  color: inherit;
}

.emotion-4 {
  margin-block-start: 0px;
  margin-block-end: 0px;
  -webkit-padding-start: 0px;
  padding-inline-start: 0px;
  padding: 0;
  list-style-type: none;
}

.emotion-5 {
  width: 100%;
  list-style: none;
}

.emotion-6 {
  font-family: 'Euclid Circular A','Helvetica Neue',Helvetica,Arial,sans-serif;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-decoration-color: transparent;
  cursor: pointer;
  font-size: inherit;
  line-height: inherit;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background: none;
  border: none;
  padding: 0;
  font-size: 13px;
  line-height: 20px;
  color: #016BF8;
  font-weight: 400;
  display: inline;
  margin: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background: none;
  border: none;
  cursor: pointer;
  position: relative;
  width: 100%;
  min-height: 32px;
  padding: 6px 16px;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: 'Euclid Circular A','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: 400;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: capitalize;
  -webkit-transition: background-color 100ms ease-in-out;
  transition: background-color 100ms ease-in-out;
  color: #001E2B;
  background-color: rgba(249,251,250,0);
  font-size: 13px;
  line-height: 20px;
  position: relative;
  padding-left: 24px;
  padding-right: 24px;
  padding-top: 8px;
  padding-bottom: 8px;
  font-size: 13px;
  line-height: 20px;
}

.emotion-6:hover,
.emotion-6:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-transition: text-decoration 150ms ease-in-out;
  transition: text-decoration 150ms ease-in-out;
  text-underline-offset: 4px;
  text-decoration-thickness: 2px;
}

.emotion-6:focus {
  outline: none;
}

.emotion-6:hover {
  text-decoration-color: #E8EDEB;
}

.emotion-6:focus {
  text-decoration-color: #016BF8;
}

.emotion-6:hover {
  background-color: #E8EDEB;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-6:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: none;
}

.emotion-6::-moz-focus-inner {
  border: 0;
}

.emotion-6:before {
  content: '';
  position: absolute;
  background-color: transparent;
  left: 0;
  top: 6px;
  bottom: 6px;
  width: 4px;
  border-radius: 0 6px 6px 0;
  -webkit-transition: -webkit-transform 150ms ease-in-out;
  transition: transform 150ms ease-in-out;
  -webkit-transform: scaleY(0.3);
  -moz-transform: scaleY(0.3);
  -ms-transform: scaleY(0.3);
  transform: scaleY(0.3);
}

.emotion-6:hover {
  background-color: #E8EDEB;
}

.emotion-6:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-6:focus:before {
  -webkit-transform: scaleY(1);
  -moz-transform: scaleY(1);
  -ms-transform: scaleY(1);
  transform: scaleY(1);
}

.emotion-6:focus {
  color: #083C90;
  background-color: #E1F7FF;
}

.emotion-6:focus:before {
  background-color: #016BF8;
}

.emotion-10 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-decoration-color: transparent;
  line-height: 13px;
  color: var(--color);
  font-weight: var(--font-weight);
}

.emotion-10>code {
  color: var(--color);
}

.emotion-10:focus,
.emotion-10:hover {
  text-decoration-line: underline;
  -webkit-transition: text-decoration 150ms ease-in-out;
  transition: text-decoration 150ms ease-in-out;
  text-underline-offset: 4px;
  text-decoration-thickness: 2px;
}

.emotion-10:focus {
  text-decoration-color: var(--focus-text-decoration-color);
  outline: none;
}

.emotion-10:hover {
  text-decoration-color: var(--hover-text-decoration-color);
}

.emotion-11 {
  margin: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background: none;
  border: none;
  cursor: pointer;
  position: relative;
  width: 100%;
  min-height: 32px;
  padding: 6px 16px;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: 'Euclid Circular A','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: 400;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: capitalize;
  -webkit-transition: background-color 100ms ease-in-out;
  transition: background-color 100ms ease-in-out;
  color: #001E2B;
  background-color: rgba(249,251,250,0);
  font-size: 13px;
  line-height: 20px;
  position: relative;
  padding-left: 24px;
  padding-right: 24px;
  padding-top: 8px;
  padding-bottom: 8px;
  font-size: 13px;
  line-height: 20px;
}

.emotion-11:hover {
  background-color: #E8EDEB;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-11:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: none;
}

.emotion-11::-moz-focus-inner {
  border: 0;
}

.emotion-11:before {
  content: '';
  position: absolute;
  background-color: transparent;
  left: 0;
  top: 6px;
  bottom: 6px;
  width: 4px;
  border-radius: 0 6px 6px 0;
  -webkit-transition: -webkit-transform 150ms ease-in-out;
  transition: transform 150ms ease-in-out;
  -webkit-transform: scaleY(0.3);
  -moz-transform: scaleY(0.3);
  -ms-transform: scaleY(0.3);
  transform: scaleY(0.3);
}

.emotion-11:hover {
  background-color: #E8EDEB;
}

.emotion-11:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-11:focus:before {
  -webkit-transform: scaleY(1);
  -moz-transform: scaleY(1);
  -ms-transform: scaleY(1);
  transform: scaleY(1);
}

.emotion-11:focus {
  color: #083C90;
  background-color: #E1F7FF;
}

.emotion-11:focus:before {
  background-color: #016BF8;
}

<li
    class="emotion-0"
  >
    <div
      class="emotion-1"
      data-testid="side-nav-group-header-label"
      id="menu-group-label-id-1"
    >
      <div
        class="emotion-2"
      >
        <div
          class="emotion-3"
        />
      </div>
    </div>
    <ul
      aria-labelledby="menu-group-label-id-1"
      class="emotion-4"
    >
      <li
        class="emotion-5"
      >
        <a
          aria-current="false"
          aria-disabled="false"
          class="lg-ui-0001 lg-ui-side-nav-item-0000 emotion-6"
          href="https://www.mongodb.com/docs/atlas/getting-started/"
          target="_self"
        >
          <span>
            Atlas
          </span>
        </a>
      </li>
      <li
        class="emotion-5"
      >
        <a
          aria-current="false"
          aria-disabled="false"
          class="lg-ui-0001 lg-ui-side-nav-item-0000 emotion-6"
          href="https://www.mongodb.com/docs/manual/"
          target="_self"
        >
          <span>
            MongoDB Database Manual
          </span>
        </a>
      </li>
      <li
        class="emotion-5"
      >
        <a
          aria-current="false"
          aria-disabled="false"
          class="emotion-10 lg-ui-side-nav-item-0000 emotion-11"
          href="/tools-and-connectors/"
          style="--color: #016BF8; --focus-text-decoration-color: #016BF8; --hover-text-decoration-color: #E8EDEB; --font-weight: inherit;"
        >
          Tools and Connectors
        </a>
      </li>
      <li
        class="emotion-5"
      >
        <a
          aria-current="false"
          aria-disabled="false"
          class="lg-ui-0001 lg-ui-side-nav-item-0000 emotion-6"
          href="https://www.mongodb.com/docs/drivers/"
          target="_self"
        >
          <span>
            Client Libraries
          </span>
        </a>
      </li>
    </ul>
  </li>
</DocumentFragment>
`;
